policybot/pkg/testresults/testresults.go

```diff
@@ -0,0 +1,337 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Take in a pr number from path "istio-prow/pr-logs/pull/istio-istio" and examine the pr
+ * for all tests that are run and their results. The results are then written to Spanner.
+ */
+package testresults
+
+import (
+	"context"
+	"encoding/json"
+
+	"io/ioutil"
+	"strconv"
+	"strings"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"google.golang.org/api/iterator"
+
+	store "istio.io/bots/policybot/pkg/storage"
+	"istio.io/pkg/log"
+)
+
+/*
```

Comment:
Here and elsewhere in this PR, you're using Java-style comments. In Go, the convention is to use // comments for functions with no empty line at the top or bottom. So just:

// Pull struct for the structure under...
type pull struct